documents/system-design-document.md

```diff
@@ -0,0 +1,455 @@
+);
+
+CREATE TABLE tasks (
+  id SERIAL PRIMARY KEY,
+  user_id INT NOT NULL,
+  name VARCHAR(128) NOT NULL,
+  description VARCHAR(128),
+  status VARCHAR(20) NOT NULL CHECK (status IN ('Created', 'Committed', 'In Progress', 'Completed')),
+  FOREIGN KEY (user_id) REFERENCES users(id)
+);
+
+
+CREATE TABLE tags (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(64) NOT NULL
+);
+
+CREATE TABLE tasks_tags (
+  task_id INTEGER NOT NULL,
+  tag_id INTEGER NOT NULL,
+  PRIMARY KEY (task_id, tag_id),
+  KEY (tag_id)
+  FOREIGN KEY (task_id) REFERENCES tasks(id),
+  FOREIGN KEY (tag_id) REFERENCES tags(id)
+);
+```
+
+## 6. API Document
+```yaml
+POST /api/users/:
+  summary: Creates a new user
+  requestBody:
+    content:
+      application/json:
+        schema:
+          type: object
+          properties:
+            id:
+              type: string
+              description: The account ID for the new user
+            password:
+              type: string
+              description: The password for the new user
+          required:
+            - account ID
+            - password
+  responses:
+    '201':
+      description: User account successfully created
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              success:
+                type: boolean
+                example: true
+    '400':
+      description: Bad Request - Invalid input
+    '409':
+      description: Conflict - Email already exists
+
+DELETE /api/users/{user_id}:
+  summary: Deletes a specific user by their ID and all associated tasks
+  parameters:
+    - name: user_id
+      in: path
+      required: true
+      schema:
+        type: integer
+      description: The unique identifier of the user to delete
+  responses:
+    '204':
+      description: No Content - User and associated tasks successfully deleted
+    '404':
+      description: User not found
+    '400':
+      description: Bad request, possibly due to an invalid user ID
+
+POST /api/tasks:
+  summary: Creates a new task for the current user, optionally associating it with description and tags
+  requestBody:
+    content:
+      application/json:
+        schema:
+          type: object
+          properties:
+            name:
+              type: string
+              description: The name of the task
+            description:
+              type: string
+              description: A detailed description of the task
+            tags:
+              type: array
+              description: A list of tag IDs to associate with the task
+              items:
+                type: integer
+          required:
+            - name
+  responses:
+    '201':
+      description: Task successfully created
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              id:
+                type: integer
+                description: The unique identifier of the newly created task
```

Comment:
What about creating a separate `Task` object, and use it for request and response payload? Returning a created object for the request would be beneficial for the client.